build: remove ENABLE_COMPILER_SUGGESTIONS option

dundargoc opened the pull request:
We need higher level and more meaningful cmake options. Having a cmake
option for only controlling compiler flags is probably not worth it as
the same can be achieved with:

cmake -B build -D CMAKE_C_FLAGS=<compiler option>

dundargoc committed:
build: remove ENABLE_COMPILER_SUGGESTIONS option

We need higher level and more meaningful cmake options. Having a cmake
option for only controlling compiler flags is probably not worth it as
the same can be achieved with:

cmake -B build -D CMAKE_C_FLAGS=<compiler option>

justinmk commented:
The intention of https://github.com/neovim/neovim/issues/343 was to have a collection of flags that were "known" to be useful (non-noisy) on the Nvim codebase.

https://github.com/neovim/neovim/pull/17815 added these flags to the cmake code and showed some related changes. But with this PR, it seems like we would need to re-discover these flags and find out which ones are useful.

> the same can be achieved with:
> 
> cmake -B build -D CMAKE_C_FLAGS=

that doesn't seem equivalent because there are numerous flags, but only some of them are useful. We stored the knowledge of the useful flags in our cmake code...

dundargoc commented:
> The intention of #343 was to have a collection of flags that were "known" to be useful (non-noisy) on the Nvim codebase.

None of the flags I removed were useful or non-noisy, that's why they were behind an option and turned off by default.

> #17815 added these flags to the cmake code and showed some related changes. But with this PR, it seems like we would need to re-discover these flags and find out which ones are useful.

None of the flags I removed were useful

> That doesn't seem equivalent because there are numerous flags, but only some of them are useful. We stored the knowledge of the useful flags in our cmake code...

None of the flags were useful.

> That misses the benefit of check_c_compiler_flag

None of the flags were useful.

dundargoc commented:
@justinmk how would you prefer to resolve this? We can always revert this PR if nothing else.

I think if a compiler warning is useful enough, then it should always be enabled. Having it behind and option just ensures no one will ever see it.

If we do enable it I think we need to add a GCC release check on CI, which is when these warnings absolutely blow up the terminal with warnings.

dundargoc commented:
I have reverted it: https://github.com/neovim/neovim/pull/22265
